src/app/backend/resource/scope/scope.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scope
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/kubernetes/dashboard/src/app/backend/resource/deployment"
+	client "k8s.io/client-go/kubernetes"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Scope structure contains two fields, deployed indicating whether scope has been deployed
+// onto the cluster, and address which is the publicly accessible address of scope.
+type Scope struct {
+	Deployed bool   `json:"deployed"`
+	Address  string `json:"address"`
+}
+
+// PostScope deploys Scope onto the cluster.
+func PostScope(client client.Interface) (*Scope, error) {
+	log.Print("Installing scope on the cluster")
+
+	response, err := http.Get("https://cloud.weave.works/launch/k8s/weavescope.yaml?k8s-service-type=NodePort")
```

Comment:
This is the part I'm worried about. It allows cloud.weave.works to hack user's cluster. Can we embed the YAML in repository here and versionize it? It'd be a bit safer at least.

Reply:
@bryk I believe that's been resolved now. Could you please review the rest of this PR and let @vlal and myself know if you broadly approve of it before we rebase and update? Thanks!